src/uniprotkb/components/entry/CommunityCuration.tsx

```diff
@@ -4,21 +4,171 @@ import {
 
+import ORCIDiDLogo from '../../../images/ORCIDiD_icon.png';
+
 import { processUrlTemplate } from '../../../shared/utils/xrefs';
 import externalUrls from '../../../shared/config/externalUrls';
 
-import { Reference } from '../../../supporting-data/citations/adapters/citationsConverter';
+import {
+  Citation,
+  CommunityAnnotation,
+  Reference,
+} from '../../../supporting-data/citations/adapters/citationsConverter';
 import EntrySection from '../../types/entrySection';
+import { DatabaseInfoMaps } from '../../utils/database';
 
 import styles from './styles/community-curation.module.scss';
 
-import ORCIDiDLogo from '../../../images/ORCIDiD_icon.png';
+const annotationGetter = (section: EntrySection) => {
+  switch (section) {
+    case EntrySection.Function:
+      return (communityAnnotation?: CommunityAnnotation) =>
+        communityAnnotation?.function;
+    case EntrySection.DiseaseVariants:
+    case EntrySection.PhenotypesVariants:
+      return (communityAnnotation?: CommunityAnnotation) =>
+        communityAnnotation?.disease;
+    case EntrySection.NamesAndTaxonomy:
+      return (communityAnnotation?: CommunityAnnotation) =>
+        communityAnnotation?.proteinOrGene;
+    default:
+      return null;
+  }
+};
+
+const groupByCommunityAnnotation = (
+  section: EntrySection,
+  communityReferences: Reference[]
+) => {
+  const getAnnotation = annotationGetter(section);
+  if (!getAnnotation) {
+    return null;
+  }
+  const annotationToCommunityReferences = new Map<string, Reference[]>();
+  for (const communityReference of communityReferences) {
+    const annotation = getAnnotation(communityReference.communityAnnotation);
+    if (annotation) {
+      const r = annotationToCommunityReferences.get(annotation) || [];
+      r.push(communityReference);
+      annotationToCommunityReferences.set(annotation, r);
+    }
+  }
+  return annotationToCommunityReferences;
+};
+
+const SubmissionDate = ({
+  accession,
+  citationId,
+  submissionDate,
+}: {
+  accession: string;
+  citationId?: Citation['id'];
+  submissionDate?: string;
+}) => (
+  <ExternalLink
+    url={
+      citationId && citationId.match(/\d+/)
+        ? externalUrls.CommunityCurationGetByAccessionAndPmid(
+            accession,
+            citationId
+          )
+        : externalUrls.CommunityCurationGetByAccession(accession)
+    }
+  >
+    <time dateTime={submissionDate}>{submissionDate}</time>
+  </ExternalLink>
+);
+
+const GroupedCommunityReference = ({
+  annotation,
+  accession,
+  references,
+  section,
+  databaseInfoMaps,
+}: {
+  annotation: string;
+  accession: string;
+  references: Reference[];
+  section: EntrySection;
+  databaseInfoMaps: DatabaseInfoMaps | null;
+}) => (
+  <Card className={styles['reference-card']}>
+    <h4>
+      {section === EntrySection.NamesAndTaxonomy
+        ? 'Community suggested name: '
+        : ''}
+      {annotation}
+    </h4>
```

Comment:
I'm wondering if we should keep the actual annotation out of the heading.
For the example of names it's fine, but when looking at functions it can get quite big and feel a bit weird. See examples (O94966 and A6NCS4):
<img width="1433" alt="Screenshot 2024-11-15 at 18 04 38" src="https://github.com/user-attachments/assets/8dab5074-1849-4cf5-b3d5-8c8229735415">
<img width="958" alt="Screenshot 2024-11-15 at 18 04 55" src="https://github.com/user-attachments/assets/9d626ada-95f2-467e-812f-a30ac8830890">

Should we have "Community suggested name:" for the names and taxonomy section and "Community annotation:" for anything else? And then have the actual annotation in a paragraph

Reply:
Nice suggestions - implemented and now looks like this:
![Screenshot 2024-11-18 at 11 40 26](https://github.com/user-attachments/assets/3c19a3cc-04b6-4c41-a4dc-0ca92a1fed28)
![Screenshot 2024-11-18 at 11 40 12](https://github.com/user-attachments/assets/9607407e-3234-4caa-b876-cf1850577e18)
![Screenshot 2024-11-18 at 11 39 45](https://github.com/user-attachments/assets/c7549062-0d5b-4eb5-a7a7-448e0d5fdb1d)